### PR TITLE
Add Shadow worker

### PR DESCRIPTION
**Pull request by yux0**
Add Shadow worker.

### CODE REVIEW

**Review by yycptt — APPROVED** [review]
LGTM

File: src/main/java/com/uber/cadence/worker/ShadowingWorker.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        new SyncActivityWorker(
+            client.getService(), client.getOptions().getDomain(), this.taskList, activityOptions);
+    activityWorker.setActivitiesImplementation(scanActivity, replayActivity);
+  }
+
+  void start() throws Exception {
+    if (!started.compareAndSet(false, true)) {
+      return;
+    }
+    startShadowingWorkflow();
```
Comment (yycptt) [inline src/main/java/com/uber/cadence/worker/ShadowingWorker.java:93]: shall we validate shadow options here? e.g. make sure exit condition is set when using continuous mode.

File: src/main/java/com/uber/cadence/worker/ShadowingWorker.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            .setDomain(shadowingOptions.getDomain())
+            .setConcurrency(shadowingOptions.getConcurrency())
+            .setExitCondition(shadowingOptions.getExitCondition())
+            .setShadowMode(shadowingOptions.getShadowMode())
+            .setSamplingRate(shadowingOptions.getSamplingRate())
+            .setTaskList(taskList)
+            .setWorkflowQuery(query);
+    StartWorkflowExecutionRequest request =
+        new StartWorkflowExecutionRequest()
+            .setDomain(shadowingOptions.getDomain())
```
Comment (yycptt) [inline src/main/java/com/uber/cadence/worker/ShadowingWorker.java:168]: this should be our system domain

**Review by yycptt — CHANGES_REQUESTED** [review]

File: src/main/java/com/uber/cadence/worker/ShadowingWorker.java (hunk 2 above)
Comment (yux0) [reply src/main/java/com/uber/cadence/worker/ShadowingWorker.java]: Good catch. updated

File: src/main/java/com/uber/cadence/worker/ShadowingWorker.java (hunk 1 above)
Comment (yux0) [reply src/main/java/com/uber/cadence/worker/ShadowingWorker.java:93]: Added into options build()

**Review by yycptt — APPROVED** [review]

**Pull request merged**